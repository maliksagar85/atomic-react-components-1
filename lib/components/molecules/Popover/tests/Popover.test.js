import React from 'react';
import { shallow, mount } from 'enzyme';

import { PopoverVanilla } from '../index';

const interactivePopover = {
  className: 'example',
  popOverHeader: 'This is a popover header text',
  trigger: <button id="popover-trigger">Click to Open</button>,
};

describe('<Popover /> Rendering', () => {
  let PopoverComponent;
  beforeEach(() => {
    PopoverComponent = shallow(<PopoverVanilla>Test</PopoverVanilla>);
  });

  test('should render correctly', () => {
    expect(PopoverComponent).toMatchSnapshot();
  });
});

describe.skip('Popover functional behavior', () => {
  let PopoverComponent;

  test('Should have Close Button By Default and can be closed', () => {
    PopoverComponent = mount(
      <PopoverVanilla {...interactivePopover} isVisible>
        Test
      </PopoverVanilla>
    );
    expect(PopoverComponent.find('.popover__close').length).toEqual(1);
    PopoverComponent = mount(
      <PopoverVanilla {...interactivePopover} hidePopoverCloseBtn>
        Test
      </PopoverVanilla>
    );
    expect(PopoverComponent.find('.popover__close').length).toEqual(0);
  });

  test('Should open By Default when isVisible set to true', () => {
    PopoverComponent = mount(
      <PopoverVanilla {...interactivePopover} isVisible>
        Test
      </PopoverVanilla>
    );
    expect(PopoverComponent.find('.popover__header').length).toEqual(1);
  });

  test('Should open and close on Trigger and X button click', () => {
    PopoverComponent = mount(
      <div>
        <p className="test">Test Content</p>
        <PopoverVanilla {...interactivePopover}>Test</PopoverVanilla>
      </div>
    );
    // Test Default Open through trigger
    PopoverComponent.find('#popover-trigger').simulate('click');
    expect(PopoverComponent.find('.popover__header').length).toEqual(1);
    // Test Close Popover when clicking outside through trigger the popover
    // Test Close through X button
    PopoverComponent.find('.popover__close').simulate('click');
    expect(PopoverComponent.find('.popover__header').length).toEqual(0);
  });
});
