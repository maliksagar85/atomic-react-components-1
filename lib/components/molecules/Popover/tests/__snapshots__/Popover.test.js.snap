// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popover /> Rendering should render correctly 1`] = `
<div
  className="popover-wrap"
  role="presentation"
/>
`;
